ConsentViewController/Classes/ConsentViewController.swift

```diff
@@ -449,23 +449,21 @@ import Reachability
     private func handleMessage(withName name: String, andBody body: [String:Any?]) {
         switch name {
         case "onReceiveMessageData": // when the message is first loaded
-            onReceiveMessage(willShow: body["willShowMessage"] as! Int == 1)
+            guard let willShow = body["willShowMessage"] as? Int else { fallthrough }
+            onReceiveMessage(willShow: willShow == 1)
         case "onMessageChoiceSelect": // when a choice is selected
-            onMessageChoiceSelect(choiceType: body["choiceType"] as! Int)
+            guard let choiceType = body["choiceType"] as? Int else { fallthrough }
+            onMessageChoiceSelect(choiceType: choiceType)
         case "interactionComplete": // when interaction with message is complete
             guard
                 let euconsent = body["euconsent"] as? String,
                 let consentUUID = body["consentUUID"] as? String
-            else {
-                print("Could not get EUConsent and ConsentUUID")
-                done()
-                return
-            }
+            else { fallthrough }
             onInteractionComplete(euconsent: euconsent, consentUUID: consentUUID)
         case "onErrorOccurred":
             onErrorOccurred(errorType: body["errorType"] as? String ?? "")
         default:
-            print("userContentController was called but the message body: \(name) is unknown.")
+            print("Couldn't parse message in userContentController. Called with name: \(name) and body: \(body)")
```

Comment:
I don't think this case will ever be called now, as the `onErrorOccurred` case above doesn't have any guard - probably isn't a problem, but `onErrorOccurred` will be called if it's failed the other cases, and will contain no error message if it can't find `body["errorType"]`.

Are you okay with that, or do you want to ensure it can get through to `default`?

Reply:
Not really @kavikumra . If there's no `body["errorType"]` then it'd be evaluated to "" (as its default ?? value) and when looking up for that error type, we'd end up with `PrivacyManagerUnknownError`. Either way, I've added an extra error for the default case and when the message name and body can't be parsed (https://github.com/SourcePointUSA/ios-cmp-app/pull/23/files#diff-76246d3a966ad7ce8b0bd0f75d46a94eR477)